src/rules/no-duplicates.js

```diff
@@ -2,21 +2,198 @@ import resolve from 'eslint-module-utils/resolve'
 import docsUrl from '../docsUrl'
 
 function checkImports(imported, context) {
-  for (let [module, nodes] of imported.entries()) {
-    if (nodes.size > 1) {
-      for (let node of nodes) {
-        context.report(node, `'${module}' imported multiple times.`)
+  for (const [module, nodes] of imported.entries()) {
+    if (nodes.length > 1) {
+      const message = `'${module}' imported multiple times.`
+      const [first, ...rest] = nodes
+      const sourceCode = context.getSourceCode()
+      const fix = getFix(first, rest, sourceCode)
+
+      context.report({
+        node: first.source,
+        message,
+        fix, // Attach the autofix (if any) to the first import.
+      })
+
+      for (const node of rest) {
+        context.report({
+          node: node.source,
+          message,
+        })
+      }
+    }
+  }
+}
+
+function getFix(first, rest, sourceCode) {
+  // Sorry ESLint <= 3 users, no autofix for you.
```

Comment:
how is this handled elsewhere in the codebase that might use `getCommentsBefore`?

Reply:
`src/ExportMap.js` does this:

```js
      // n.leadingComments is legacy `attachComments` behavior
      if ('leadingComments' in n) {
        leadingComments = n.leadingComments
      } else if (n.range) {
        leadingComments = source.getCommentsBefore(n)
      }
```

However, I tried something like this and that at least makes the autofix not crash, but instead the autofix did nothing. As far as I know, ESLint <= 3 does not support multiple `fixer.whatever()` in the `fix`, which is required for `no-duplicates` – we both update the first import and remove one or more others. I can add a comment explaining all of this.

Reply:
Comment added.